Feature/lit 2726 Lit Action Plus Documentation (#274)
LIT-Protocol/docs: docs/sdk/serverless-signing/combining-decryption-shares.md

```js
 const chain = 'ethereum';
  const accessControlConditions = [
    {
      contractAddress: '',
      standardContractType: '',
      chain,
      method: 'eth_getBalance',
      parameters: [':userAddress', 'latest'],
      returnValueTest: {
        comparator: '>=',
        value: '0',
      },
    },
  ];
  const message = 'Hello world';
  const client = new LitNodeClient({
    network: 'cayenne'
  });
  await client.connect();
  const { ciphertext, dataToEncryptHash } = await LitJsSdk.encryptString(
    {
      accessControlConditions,
      authSig: globalThis.LitCI.CONTROLLER_AUTHSIG,
      chain,
      dataToEncrypt: message,
    },
    client
  );

  console.log("cipher text:", ciphertext, "hash:", dataToEncryptHash);
```

Let's now take the `cipther text` and `hash of encrypted data` and use it from a Lit Action to decrypt within the `TEE`.

```js
const code = `(async () => {
  const resp = await Lit.Actions.decryptAndCombine({
    accessControlConditions,
    ciphertext,
    dataToEncryptHash,

>     authSig,

Author: Note, supply an example authSig as `jsParams` as we cannot get the authSig context in the lit action scope.

Author: The above note is now addressed through `authSig` implicit loading done by the `decryptAndCombine` op code binding.

import FeedbackComponent from "@site/src/pages/feedback.md";

# Decrypting and Combining Within an Action

## Overview

> Within a Lit action, you may choose to combine a `decryption shares` from within the context of an action. Meaning signature shares from each node will be combined and given to every node which recieved a request to execute the given action. Combining within an action may be useful if you wish to take advantage of the `Secure Compute Enviorment` offered by the Lit network. Actions which take combine signatures from within their own execution context will no provide the shares of the signatures to the client. So all information will stay in the trusted execution enviorment (TEE).

Reviewer: Copy suggestion:

Current: 
"Within a Lit action, you may choose to combine a `decryption shares` from within the context of an action. Meaning signature shares from each node will be combined and given to every node which recieved a request to execute the given action. Combining within an action may be useful if you wish to take advantage of the `Secure Compute Enviorment` offered by the Lit network. Actions which take combine signatures from within their own execution context will no provide the shares of the signatures to the client. So all information will stay in the trusted execution enviorment (TEE)."

Suggestion: 
"[Decryption with Lit](https://developer.litprotocol.com/v3/sdk/access-control/quick-start) is typically done client-side by an authorized party at the time of access. The `decryptAndCombine` function allows you to decrypt data within a Lit Action. This function is useful for performing operations over sensitive data, where the data itself remains private within the confines of each [Lit node's TEE](https://developer.litprotocol.com/v3/resources/how-it-works#1-lit-nodes). 

When you call `decryptAndCombine`, the `decryption shares` from each node are collected and aggregated on a single Lit node before they are combined. "

^^ that function is only run on a single node right?

Reviewer 2: @a1ttech - currently the decryption shares are provided to every node, and each recombines them.   This means that ALL nodes have a copy of the recombined data, rather than just one node.

# Encrypting content
We will start by performing an `encrypt` operation as shown below using the `LitNodeClient` this operation is entirely done on the client, so no need for any lit action invovelment.
```js
 const chain = 'ethereum';

>   const accessControlConditions = [

Reviewer 3: Can we correct spacing here?

    {
      contractAddress: '',
      standardContractType: '',
      chain,
      method: 'eth_getBalance',
      parameters: [':userAddress', 'latest'],
      returnValueTest: {
        comparator: '>=',
        value: '0',
      },
    },
  ];
  const message = 'Hello world';
  const client = new LitNodeClient({
    network: 'cayenne'
  });
  await client.connect();

>   // ==================== Test Logic ====================

Reviewer 3: I think we can remove this comment

  const { ciphertext, dataToEncryptHash } = await LitJsSdk.encryptString(
    {
      accessControlConditions,

>       authSig: globalThis.LitCI.CONTROLLER_AUTHSIG,

Reviewer 3: I think we can simply write:
```js
authSig,
```
as `globalThis.LitCI.CONTROLLER_AUTHSIG` will not be known to our users

Author: yes sorry this is some left over from the local setup environment

import FeedbackComponent from "@site/src/pages/feedback.md";

# Decrypting and Combining Within an Action

## Overview

> Decryption with Lit is typically done client-side by an authorized party at the time of access. The decryptAndCombine function allows you to decrypt data within a Lit Action. This function is useful for performing operations over sensitive data, where the data itself remains private within the confines of each Lit node's TEE.

Reviewer 4: ```suggestion
Decryption with Lit is typically done client-side by an authorized party at the time of access. The decryptAndCombine function allows you to decrypt data within a Lit Action. This function is useful for performing operations over sensitive data, where the data itself remains private within the confines of each Lit node's trusted execution environment (TEE).
```

> When you call decryptAndCombine, the decryption shares are collected from each Lit node before they are combined on a single node.

Reviewer 4: ```suggestion
When you call `decryptAndCombine`, the decryption shares are collected from each Lit node before they are combined on a single node.
```

# Encrypting content

> We will start by performing an `encrypt` operation as shown below using the `LitNodeClient` this operation is entirely done on the client, so no need for any lit action invovelment.

Reviewer 4: ```suggestion
We will start by performing an `encrypt` operation as shown below using the `LitNodeClient` this operation is entirely done on the client, so no need for any Lit Action involvement.
```

```js
 const chain = 'ethereum';
 const accessControlConditions = [
    {
      contractAddress: '',
      standardContractType: '',
      chain,
      method: 'eth_getBalance',
      parameters: [':userAddress', 'latest'],
      returnValueTest: {
        comparator: '>=',
        value: '0',
      },
    },
  ];
  const message = 'Hello world';
  const client = new LitNodeClient({
    network: 'cayenne'
  });
  await client.connect();
  const { ciphertext, dataToEncryptHash } = await LitJsSdk.encryptString(
    {
      accessControlConditions,
      sessionSigs: {}, // your session
      chain,
      dataToEncrypt: message,
    },
    client
  );

  console.log("cipher text:", ciphertext, "hash:", dataToEncryptHash);
```

> Let's now take the `cipther text` and `hash of encrypted data` and use it from a Lit Action to decrypt within the `TEE`.

Reviewer 4: ```suggestion
Let's now take the `ciphertext` and `dataToEncryptHash` and use it from a Lit Action to decrypt within the TEE.
```

In the below example we set the `authSig` to `null` as a way to tell the Lit Action runtime to use the `authSig` which was provided to the node through the `executeJs` call's `sessionSigs`.
If you wish you may provide a different Auth Signature if the one provided from the session is not relevant to your use case. 
```js
const code = `(async () => {
  const resp = await Lit.Actions.decryptAndCombine({
    accessControlConditions,
    ciphertext,
    dataToEncryptHash,
    authSig: null,
    chain: 'ethereum',
  });

  Lit.Actions.setResponse({ response: resp });
})();`

> const res = client.executeJs({

Reviewer 4: ```suggestion
const res = await client.executeJs({
```
